markymark/Classes/Rules/Inline/LinkRule.swift

```diff
@@ -12,18 +12,20 @@ open class LinkRule: InlineRegexRule {
     /// Example: [Google](http://www.google.com "with custom title")
     open var expression = NSRegularExpression.expressionWithPattern(
         //             [  title  ] (    URL   "     optional title    "   )
-        #"(?<!!\p{Z}?)\[{1}(.+?)\]\({1}(.+?)( "[[:print:]^"]+")?\)"#
+        #"(?<!!\p{Z}?)\[{1}(.+?)\]\({1}([^ ]+?)((?: "(.+?)[\")])?)?\)"#
     )
 
     // MARK: Rule
 
     open func createMarkDownItemWithLines(_ lines: [String]) -> MarkDownItem {
+        let title: String? = lines.first?.optionalSubStringWithExpression(expression, ofGroup: 4)
```

Comment:
`String?` can be removed

Reply:
Good spot, a lot more elements within the project could have their Type removed.
I will note this down as a more general refactor to be performed in the future.